src/test/scala/io/iohk/ethereum/security/SSLContextFactorySpec.scala

```diff
@@ -0,0 +1,224 @@
+package io.iohk.ethereum.security
+
+import java.io.{ByteArrayInputStream, File, FileInputStream, FileOutputStream}
+import java.security.{KeyStore, SecureRandom}
+
+import javax.net.ssl.{KeyManager, TrustManager}
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.io.BufferedSource
+
+class SSLContextFactorySpec extends AnyFlatSpec with Matchers with MockFactory with BeforeAndAfterAll {
+
+  val fileName: String = "temp.txt"
+  var file: File = _
+
+  override def beforeAll {
+    new FileOutputStream(fileName, false).getFD
+    file = new File(fileName)
+  }
+
+  override def afterAll {
+    file.delete()
+  }
+
+  val keyStorePath = "mantisCA.p12"
+  val keyStoreType = "pkcs12"
+  val passwordFile = "password"
+
+  it should "createSSLContext" in new TestSetup(
+    List(keyStorePath, passwordFile),
+    () => Right(new FileInputStream(file)),
+    () => Right(()),
+    () => Right(Array.empty),
+    () => Right(Array.empty)
+  ) {
+
+    val sslConfig = SSLConfig(
+      keyStorePath = keyStorePath,
+      keyStoreType = keyStoreType,
+      passwordFile = passwordFile
+    )
+    sSLContextFactory.createSSLContext(sslConfig, new SecureRandom()) match {
+      case Right(ssl) =>
+        ssl.getProtocol shouldBe "TLS"
+      case Left(error) => fail(error.reason)
+    }
+  }
+
+  it should "return a Error because keystore path and password are missing" in new TestSetup(
+    Nil,
+    () => Right(new FileInputStream(file)),
+    () => Right(()),
+    () => Right(Array.empty),
+    () => Right(Array.empty)
+  ) {
+
+    val sslConfig = SSLConfig(
+      keyStorePath = keyStorePath,
+      keyStoreType = keyStoreType,
+      passwordFile = passwordFile
+    )
+    val response = sSLContextFactory.createSSLContext(sslConfig, new SecureRandom())
+    response shouldBe Left(SSLError("Certificate keystore path and password file configured but files are missing"))
+  }
+
+  it should "return a Error because keystore path is missing" in new TestSetup(
+    List(passwordFile),
+    () => Right(new FileInputStream(file)),
+    () => Right(()),
+    () => Right(Array.empty),
+    () => Right(Array.empty)
+  ) {
+
+    val sslConfig = SSLConfig(
+      keyStorePath = keyStorePath,
+      keyStoreType = keyStoreType,
+      passwordFile = passwordFile
+    )
+    val response = sSLContextFactory.createSSLContext(sslConfig, new SecureRandom())
+    response shouldBe Left(SSLError("Certificate keystore path configured but file is missing"))
+  }
+
+  it should "return a Error because password file is missing" in new TestSetup(
+    List(keyStorePath),
```

Comment:
```suggestion
          existingFiles = List(keyStorePath),
   ....
```
^ for every param, for every case...

Reply:
Done